### PR TITLE
Santanu crawler

### PR DESCRIPTION
**Pull request by ghost**
saving tweet streams.

### CODE REVIEW

File: crawler/twitter.js [hunk 1]
```diff
@@ -4,9 +4,12 @@ const cfg       = require('../core/config'),
       debug     = require('../core/debug')('crawler/twitter', true),
       Character = require('../models/character'),
       Tweet     = require('../models/tweet'),
-      Twit      = require('twit');
+      Twit      = require('twitter'),
```
Comment (julienschmidt) [inline crawler/twitter.js:7]: Please don't change the twitter client back. I changed it for a reason. Neither of both is an official package, but the latter is better maintained.

File: crawler/twitter.js [hunk 2]
```diff
@@ -4,9 +4,12 @@ const cfg       = require('../core/config'),
...
+      got       = require('../crawler/got'),
```
Comment (julienschmidt) [inline crawler/twitter.js:8]: you aren't using this package

File: crawler/twitter.js [hunk 3]
```diff
@@ -209,13 +215,28 @@ twitter.crawlAll = function(full) {
...
+    output = "'" + output.substring(0, output.length - 2) + "'";
+    console.log("streaming...");
     client.stream('statuses/filter', {
-        track: query
+        track: output
     }, function(stream) {
-        stream.on('data', twitter.saveTweet);
-
+        stream.on('data', function(data) {
+            var lcString = util.inspect(data.text).toLowerCase();
```
Comment (julienschmidt) [inline crawler/twitter.js:231]: What do you need `util.inspect` for?

File: example/app/app.js [hunk 4]
```diff
@@ -33,6 +33,7 @@ app.get('/', function(req, resp) {
     ps[0] = gotsent.mostPopular();
     ps[1] = gotsent.mostHated();
     ps[2] = gotsent.mostDiscussed();
+    gotsent.startStream();
```
Comment (julienschmidt) [inline example/app/app.js:36]: You certainly don't want to start a new stream every time someone visits the main page.

File: index.js [hunk 5]
```diff
@@ -33,7 +33,7 @@ pkg.cfg = cfg;
 pkg.init = function() {
     db.connect();
     twitter.connect();
-    //pkg.update();
+    pkg.update();
```
Comment (julienschmidt) [inline index.js:36]: This doesn't belong here

File: crawler/twitter.js [hunk 6]
```diff
@@ -32,6 +35,9 @@ twitter.getTweetsList = function(ids, retries) {
         ids = ids.toString();
     }
 
+
+exports.fetchTweets = function(id, query, maxID) {
```
Comment (julienschmidt) [inline crawler/twitter.js:39]: Your branch is still corrupt. Please start a clean branch from the current `develop` like I already said.

File: crawler/twitter.js (hunk 1 above)
Comment (ghost) [inline crawler/twitter.js:7]: ok

**Comment by julienschmidt** [comment]
Also: use `npm test`

File: crawler/twitter.js [hunk 7]
```diff
@@ -209,13 +215,28 @@ twitter.crawlAll = function(full) {
...
     client.stream('statuses/filter', {
-        track: query
+        track: output
     }, function(stream) {
-        stream.on('data', twitter.saveTweet);
-
+        stream.on('data', function(data) {
+            var lcString = util.inspect(data.text).toLowerCase();
+            for (var i = 0; i < query.length; i++) {
+                var lcSearchString = query[i].name.toLowerCase();
```
Comment (julienschmidt) [inline crawler/twitter.js:233]: I'm also not so sure if it is efficient enough to compare against > 2000 character in a linear way. I'd expect quite a lot of tweets during / after the episodes.
The streaming also doesn't allow "strict matching" as it seems. So we will also get a lot of strings which don't match anything.

If we are too slow Twitter will disconnect us. The reconnect logic is also currently missing.

**Comment by julienschmidt** [comment]
Even with all this fixed, there is still one major problem: We might have the tweets in our DB then, but they're not in the generated CSV then. Regenerating the CSV with every single Tweet is not an option.

**Comment by ghost** [comment]
this PR will not be used. please refer to new PR, stream

**Pull request closed without merging**